drivers/temperature/ltc2983/ltc2983.c

```diff
@@ -0,0 +1,644 @@
+int ltc2983_chan_read_raw(struct ltc2983_desc *device, const int chan,
+			  uint32_t *val)
+{
+	uint32_t start_conversion = 0;
+	uint16_t reg_addr = LTC2983_CHAN_RES_ADDR(chan);
+	uint8_t raw_array[] = {LTC2983_SPI_READ_BYTE, reg_addr >> 8,
+			       reg_addr & 0xFF, 0, 0, 0, 0
+			      };
+	int ret;
+
+	start_conversion = LTC2983_STATUS_START(true);
+	start_conversion |= LTC2983_STATUS_CHAN_SEL(chan);
+	/* start conversion */
+	ret = ltc2983_reg_write(device, LTC2983_STATUS_REG, start_conversion);
+	if (ret)
+		return ret;
+
+	no_os_mdelay(300);
+
+	/* read the converted data */
+	ret = no_os_spi_write_and_read(device->comm_desc, raw_array,
+				       NO_OS_ARRAY_SIZE(raw_array));
+	if (ret)
+		return ret;
+
+	*val = (raw_array[3] << 24) + (raw_array[4] << 16) +
+	       (raw_array[5] << 8) + raw_array[6];
+
+	if (!(LTC2983_RES_VALID_MASK & *val))
+		return -EIO;
+
+	if (device->sensors[chan]->type <= LTC2983_THERMOCOUPLE_CUSTOM)
+		ret = ltc2983_thermocouple_fault_handler(*val);
+	else
+		ret = ltc2983_common_fault_handler(*val);
+	if (ret)
+		return ret;
+
+	*val = no_os_sign_extend32((*val) & LTC2983_DATA_MASK,
+				   LTC2983_DATA_SIGN_BIT);
+	return 0;
+}
+
+/**
+ * @brief Set scale of raw channel data / temperature
+ * @param device - LTC2983 descriptor
+ * @param chan - channel number
+ * @param val - scale numerator
+ * @param val2 - scale denominator
+ * @return 0 in case of success, errno errors otherwise
+ */
+int ltc2983_chan_read_scale(struct ltc2983_desc *device, const int chan,
+			    uint32_t *val, uint32_t *val2)
+{
+	if (device->sensors[chan]->type == LTC2983_DIRECT_ADC) {
+		/* value in millivolt */
+		*val = 1000;
+		/* 2^21 */
+		*val2 = 2097152;
+	} else {
+		/* value in milli degrees */
+		*val = 1000;
+		/* 2^10 */
+		*val2 = 1024;
+	}
+	return 0;
+}
+
+/**
+ * @brief Channel assignment for common register fields
+ * @param device - LTC2983 descriptor
+ * @param sensor - LTC2983 common sensor descriptor
+ * @param chan_val - channel assignment value to be written
+ * @return 0 in case of success, errno errors otherwise
+ */
+int __ltc2983_chan_assign_common(struct ltc2983_desc *device,
+				 const struct ltc2983_sensor *sensor,
+				 uint32_t chan_val)
+{
+	uint32_t reg = LTC2983_CHAN_START_ADDR(sensor->chan);
+
+	chan_val |= LTC2983_CHAN_TYPE(sensor->type);
+	uint8_t raw_array[] = {LTC2983_SPI_WRITE_BYTE, (reg >> 8) & 0xFF,
+			       reg & 0xFF, (chan_val >> 24) & 0xFF,
+			       (chan_val >> 16) & 0xFF, (chan_val >> 8) & 0xFF,
+			       chan_val & 0xFF
+			      };
+
+	return no_os_spi_write_and_read(device->comm_desc, raw_array,
+					NO_OS_ARRAY_SIZE(raw_array));
+}
+
+/**
+ * @brief Channel assignment for custom register fields
+ * @param device - LTC2983 descriptor
+ * @param custom - LTC2983 custom sensor descriptor
+ * @param chan_val - channel assignment value to be written
+ * @return 0 in case of success, errno errors otherwise
+ */
+int __ltc2983_chan_custom_sensor_assign(struct ltc2983_desc *device,
```

Comment:
Are these functions supposed to be exported to a user? I'm asking because of the `__*` naming. If not, please make them `static`.